crawler.py

```diff
@@ -1,70 +1,93 @@
 import mechanize
-from collections import namedtuple
+import sys
+import pprint
+import collections
 from urllib.parse import urljoin, urlparse
 
-def validated(url):
-    '''Checks if given url is proper.'''
-    browser = mechanize.Browser()
-    browser.set_handle_robots(False)
-    try:
-        browser.open(url)
+
+def is_validated(url):
+    """Checks if given url is proper."""
+    parsed_url = urlparse(url)
+    if parsed_url.scheme and parsed_url.netloc:
         return True
-    except:
+    else:
         return False
 
-def validated_link(link, host_url):
-    '''Check if link is relative and make it absolute.'''
-    if '//' not in link:
-        return urljoin(host_url, link)
+
+def make_url_absolute(url, domain_full_address):
+    """Check if link is relative and make it absolute."""
+    if "//" not in url:
+        return urljoin(domain_full_address, url)
     else:
-        return link
- 
- 
-def page_properties(page_url, domain, netloc):
-    '''Returns page properties:
-        all links from specific domain and page title.'''
-    
+        return url
+
+
+def domain_tuple(url):
+    """Returns namedtuple with full domain address and domain name of given url."""
+    parsed_url = urlparse(url)
+    Domain = collections.namedtuple("Domain", "full_address, name")
+    return Domain(f"{parsed_url.scheme}://{parsed_url.netloc}", parsed_url.netloc)
+
+
+def page_title(url):
+    """Returns title of given url."""
+    browser = mechanize.Browser()
+    browser.set_handle_robots(False)
+    browser.open(url)
+    return browser.title()
+
+
+def page_links(url):
+    """Returns all links from given page."""
     browser = mechanize.Browser()
     browser.set_handle_robots(False)
```

Comment:
What does this line do?

Reply:
When I was first writing this code for recruitment I wanted to get title from external page(I don't remember what page it was) and I came across problem that I couldn't get any data, solution for this problem was this line. Now I've tested my code without this line in title and links function and everything went well.

Mechanize documentation says this line "Ignore robots.txt"